datanode/entities/market.go

```diff
@@ -133,6 +135,21 @@ func NewMarketFromProto(market *vega.Market, txHash TxHash, vegaTime time.Time)
 		quadraticSlippageFactor = &factor
 	}
 
+	parentMarketID := MarketID("")
+	if market.ParentMarketId != nil && *market.ParentMarketId != "" {
+		parent := MarketID(*market.ParentMarketId)
+		parentMarketID = parent
+	}
+
+	insurancePoolFraction := (*num.Decimal)(nil)
```

Comment:
Wouldn't that works nicer?
```suggestion
	var insurancePoolFraction *num.Decimal
```